docs/architecture/authentication-and-authorization.md

```diff
@@ -0,0 +1,125 @@
+# Authentication and Authorization
+
+## Introduction
+Currently, communication between the Compass and both Runtimes and Applications are not secured. We need to provide some security possibilities.
+We want to secure the Compass using ORY's Hydra and OathKeeper. There would be three ways of authentication:
+ - OAuth 2.0 
+ - Client Certificates (mTLS)
+ - JWT token issued by Dex
+
+ There would be three possibilities for securing the connection. To achieve that, first, we need to integrate Hydra and OathKeeper into the Compass. We also need to implement additional supporting components to make our scenarios valid.
+
+## Architecture
+
+The following diagram represents architecture of the security in Compass:
+
+![](./assets/security-architecture.svg)
+
+### Tenant mapping handler
```

Comment:
This section is a bit outdated after our latest discussions regarding creating system_auths db table and returning arrays of auths.

Reply:
Updated